components/core/src/clp_s/search/ast/SearchUtils.cpp

```
@@ -295,4 +177,146 @@ auto unescape_kql_internal(std::string const& value, std::string& unescaped, boo
+) {
+    for (auto it = child->op_begin(); it != child->op_end(); it++) {
+        auto sub_expr = std::static_pointer_cast<Expression>(*it);
+        sub_expr->set_parent(parent.get());
+    }
+    parent->get_op_list().splice(location, child->get_op_list());
+}
+
+bool double_as_int(double in, FilterOperation op, int64_t& out) {
+    switch (op) {
+        case FilterOperation::EQ:
+            out = static_cast<int64_t>(in);
```

Cppcheck (2.20.0)

[style] The function 'set_matching_types' is never used.

(unusedFunction)